From dec784b84f2b6337156b3236af22dadf310326ad Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:42:22 -0400
Subject: [ethernet] cxgb4: silence shift wrapping static checker warning

Message-id: <1410475447-94392-4-git-send-email-svemuri@redhat.com>
Patchwork-id: 93789
O-Subject: [RHEL7.1 PATCH BZ 1124947 003/108] cxgb4: silence shift wrapping static checker warning
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

I don't know how large "tp->vlan_shift" is but static checkers worry
about shift wrapping bugs here.

Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Acked-by: Dimitris Michailidis <dm@chelsio.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 9ef9730ba84d4021a27d3a1679fd50f9bac0e0e7)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/chelsio/cxgb4/l2t.c b/drivers/net/ethernet/chelsio/cxgb4/l2t.c
index cb05be9..81e8402 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/l2t.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/l2t.c
@@ -423,7 +423,7 @@ u64 cxgb4_select_ntuple(struct net_device *dev,
   * in the Compressed Filter Tuple.
   */
  if (tp->vlan_shift >= 0 && l2t->vlan != VLAN_NONE)
-  ntuple |= (F_FT_VLAN_VLD | l2t->vlan) << tp->vlan_shift;
+  ntuple |= (u64)(F_FT_VLAN_VLD | l2t->vlan) << tp->vlan_shift;
 
  if (tp->port_shift >= 0)
   ntuple |= (u64)l2t->lport << tp->port_shift;
-- 
1.7.1